tests/io/bytesio_ext2: Remove dependency on specific EINVAL value

mikewadsten opened the pull request:
If MICROPY_USE_INTERNAL_ERRNO is disabled, MP_EINVAL is not guaranteed
to have the value 22, so we cannot depend on OSError(22,).
Instead, use errno.errorcode to get the symbolic name of the error.

If it would be better to just crash in case of not throwing OSError, rather than catching and printing it, I can gladly make that change.

dpgeorge commented:
Thanks, but this now pulls in a dependency on (u)errno which could make the test unusable on certain ports that don't have this module.  Since the aim here is not to test BytesIO and not OSError we should find a way to not need (u)errno, and also not rely on the number being 22.  Maybe just print the type of the exception (should be OSError) and check that the error code is in a list of known EINVAL values?

mikewadsten commented:
To be fair, you can only run the complete `io` suite of tests on a PC port because of the inherent dependency on the data directory.

But you're right about the dependency on uerrno. Our port defines MP_EINVAL to 7022 -  do you know of any other values (aside from 22) that we should check against?

mikewadsten commented:
That weakens the assertion being made (and called out in the comment, checking for `OSError(EINVAL)` but I'll go ahead and make that change.

mikewadsten committed:
tests/io/bytesio_ext2: Remove dependency on specific EINVAL value

If MICROPY_USE_INTERNAL_ERRNO is disabled, MP_EINVAL is not guaranteed
to have the value 22, so we cannot depend on OSError(22,).
Instead, to support any given port's errno values, without relying
on uerrno, we just check that the args[0] is positive.

dpgeorge commented:
> That weakens the assertion being made

Yes but our hands are tied, not much can be done when we want to try and support an arbitrary port.

So thanks for updating, and for the good commit message; merged in 9f1eafc38092213bf59cd5dfdd803249077729c3
